### ISSUE TITLE
Nested queries are not executed

### ISSUE THREAD

**Issue by deivkk** [tinrab/spidey#3]
Hi @tinrab ,

First at all, thanks for your articles at outcrawl.com, I am personally learning a lot from them.

However I have a couple of doubts:

- Why do you take Order model out from the code generation tool? You mention that _" more control is needed for the Order mode"_ but I don't understand why.
- I am not able to retrieve any Order when querying Accounts, it always returns an empty array. I have already checked that all data from creation of Accounts, Products and Orders is in the database. Also, I have noticed that method _Account_orders_ is never called. 

I have read the docs from https://gqlgen.com/getting-started/ where the binding rules are described:

- _If there is a property with that name and type, use it_
- _If there is a method with that name and type, use it_
- _Otherwise, add it to the Resolvers interface. This is the magic._

So I am assuming that since model Account contains a property []Order it tries to resolve it without calling to Account_orders, resulting always in an empty array. Also I am totally confused on the example provided in gqlgen tutorial: both User and Todo are not pregenerated (but User would generate the same model, while Todo uses a String instead of an Object for property UserID), and the Todo_user method is added automatically to the Resolvers interface, something that is not happening in the Spidey example.

There is something that I'm missing...

Thanks!

**Comment by tinrab** [comment]
There was an update to gqlgen tool that breaks this project. I don't know exactly what happened, but I fixed it now on [0f3f94a](https://github.com/tinrab/spidey/commit/0f3f94a64760ed4125d9216e4faac6154bae997a). I've manually created Order model, because it was impossible to get its products in one RPC call. That's all I can remember, but I might've made a dumb mistake.

**Comment by tinrab** [comment]
Yes, but now the `Todo_user` method isn't even generated for the `Resolvers` struct. Same with Spidey, only top level queries are generated: `Query_accounts` and `Query_products`. `Account_orders` is missing.

**Comment by deivkk** [comment]
Maybe I've messed up with different gqlgen versions, but method `Todo_user` is still added to the `Resolvers`when running `go generate`. I will try again with a cleaner workspace
